File: app/src/main/java/com/example/pam_project/lists/categories/components/CategoryViewHolder.java
package com.example.pam_project.lists.categories.components;

import android.util.Log;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.pam_project.R;
import com.example.pam_project.lists.lists.listActivity.OnListClickedListener;

public class CategoryViewHolder extends RecyclerView.ViewHolder {

    private OnListClickedListener listener;

    public CategoryViewHolder(@NonNull View itemView) {
        super(itemView);
    }

    public void bind(final CategoryInformation category) {
        final TextView categoryTitle = itemView.findViewById(R.id.category_title);
        categoryTitle.setText(category.getTitle());

        itemView.setOnClickListener(v -> {
            if (listener != null) {
                listener.onClick(category.getId());
            }
        });
    }

    public void setOnClickListener(final OnListClickedListener listener) {
        this.listener = listener;
    }
}

File: app/src/main/java/com/example/pam_project/features/categories/list/CategoryViewHolder.kt
package com.example.pam_project.features.categories.list

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.pam_project.R
import com.example.pam_project.features.lists.list.OnListClickedListener

class CategoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private var listener: OnListClickedListener? = null
    fun bind(category: CategoryInformation?) {
        val categoryTitle = itemView.findViewById<TextView>(R.id.category_title)
        categoryTitle.text = category.getTitle()
        itemView.setOnClickListener { v: View? ->
            if (listener != null) {
                listener!!.onClick(category.getId())
            }
        }
    }

    fun setOnClickListener(listener: OnListClickedListener?) {
        this.listener = listener
    }
}